src/data/scripts/main.py

```diff
@@ -1,43 +1,125 @@
 from openthaigpt_pretraining_data.mc4.preprocess import (
-    clean_dataset as clean_mc4_dataset,
+    clean_text as clean_mc4_text,
 )
 from openthaigpt_pretraining_data.oscar.preprocess import (
-    clean_dataset as clean_oscar_dataset,
+    clean_text as clean_oscar_text,
+)
+from openthaigpt_pretraining_data.core.perplexity import (
+    classify_spam,
+    sample_text_back,
 )
-from openthaigpt_pretraining_data.core.preprocess import clean_dataset as clean_dataset
 import argparse
+import jsonlines
+from multiprocessing import Pool
+import numpy as np
+import scipy
 
 parser = argparse.ArgumentParser()
 
 parser.add_argument(
     "--input_file",
-    help='Name of an input file (Default: "input.txt")',
-    default="input.txt",
+    help='Name of an input file (Default: "scripts/input.jsonl")',
+    default="scripts/input.jsonl",
 )
 parser.add_argument(
     "--output_file",
-    help='Name of an output file (Default: "output.txt")',
-    default="output.txt",
+    help='Name of an output file (Default: "scripts/output.jsonl")',
+    default="scripts/output.jsonl",
+)
+parser.add_argument(
+    "--buffer_capacity",
+    help="""Size of the buffer.The bigger it is, the more parallelization benefits us
+            but be aware of memory consumtion. (Default: 1000)""",
+    default=1000,
 )
 parser.add_argument(
-    "--engine",
-    help='Engine of preprocessing (Default: "core")',
-    default="core",
-    choices=["core", "oscar", "mc4"],
+    "--sampled_back_ratio",
+    help="""Ratio of data classified as spam to sampled back to the dataset.
+    (Default: 0.1)""",
+    default=0.1,
 )
 
+
 args = parser.parse_args()
 
-CLEAN_FUNCTION = {
-    "mc4": clean_mc4_dataset,
-    "oscar": clean_oscar_dataset,
-    "core": clean_dataset,
-}
 
-with open(args.input_file, "r", encoding="utf-8") as f:
-    dataset = [line.strip() for line in f.readlines()]
+def clean_data(item):
+    text = item["text"]
+
+    text = text.strip()
+    text = clean_mc4_text(text)
+
+    if text == "":
+        item["text"] = text
+        return None, None, item
+
+    text = clean_oscar_text(text)
+
+    if text == "":
+        item["text"] = text
+        return None, None, item
+
+    prediction, log_pp_score = classify_spam(text)
+
+    item["text"] = text
+    return prediction, log_pp_score, item
+
+
+BUFFER_CAP = int(args.buffer_capacity)
```

Comment:
What if instead of using JSONL, we convert all internet data to huggingface dataset and use .map function to automatically do multi-processing, chunk, and buffer?